bundles/org.openhab.binding.nikobus/src/main/java/org/openhab/binding/nikobus/internal/handler/NikobusPushButtonHandler.java

```diff
@@ -177,13 +263,41 @@ public void handleCommand(ChannelUID channelUID, Command command) {
         }
     }
 
+    // called when there is a message on the Nikobus
     private void commandReceived() {
-        if (thing.getStatus() != ThingStatus.ONLINE) {
-            updateStatus(ThingStatus.ONLINE);
-        }
 
+        // a button itself does not have an on and off state
+        // so this command will be correct only half of the time...
         updateState(CHANNEL_BUTTON, OnOffType.ON);
 
+        if (logger.isDebugEnabled()) {
+            ChannelUID channelUID = new ChannelUID(this.getThing().getUID(), CHANNEL_BUTTON);
+            logger.debug("button trigger {} {}", channelUID, CommonTriggerEvents.PRESSED);
+        }
+        if (logger.isTraceEnabled()) {
+            
+            // print out the address a seen in the PC application
+            int addressPCInt = Integer.parseInt(this.address, 16);
+            String addressPCString = Integer.toBinaryString(addressPCInt);
+            // reverse bits
+            StringBuilder sb = new StringBuilder(addressPCString);
+            addressPCString = sb.reverse().toString();
+            // last three bits hold the specific button values (0-7)
+            int lastThreeBits = Integer.parseInt(addressPCString.substring(addressPCString.length() - 3), 2);
+            // transform into the letter code (A, B, 1A, ...)
+            String buttonName = Utils.getKeyByValue(buttonIndexMap, lastThreeBits);
+            // drop the last two bits, then set the last bit to 0 to get the base address
+            String baseAddress = addressPCString.substring(0, addressPCString.length() - 3) + "0";
+            int baseAddressInt = Integer.parseInt(baseAddress, 2);
+            logger.trace(" button press bus {}, pc {}:{}", this.address, String.format("%06X", baseAddressInt), buttonName);
+
```

Comment:
remove blank line

Reply:
I need impactedModulesPC for display in PaperUI